packages/core/src/StateNode.ts

```diff
@@ -371,7 +371,7 @@ class StateNode<
       if (!this.machine.options.services[resolvedSrc]) {
         this.machine.options.services = {
           ...this.machine.options.services,
-          [resolvedSrc]: invokeConfig.src
+          [resolvedSrc]: invokeConfig.src as any
```

Comment:
Not necessarily as part of this PR but I think we should remove overriding `this.machine.options` internally - from what I remember this is the last place where we do.

If we wouldnt override this we could more easily swap the config with a simple assignment rather than doing this:
https://github.com/davidkpiano/xstate/blob/be2bc0dd388901a894d6a2afd7eef67931194a11/packages/xstate-react/src/index.ts#L87-L93
which doesnt currently even cover all config types (guards and delays stay unhandled now).